diff --git a/src/main/force.F90 b/src/main/force.F90
index e6eff344..4ad2db5f 100644
--- a/src/main/force.F90
+++ b/src/main/force.F90
@@ -2632,7 +2632,7 @@ subroutine finish_cell_and_store_results(icall,cell,fxyzu,xyzh,vxyzu,poten,dt,dv
        tstop(:,i) = tstopi(:)
     elseif (use_dust .and. .not.use_dustfrac) then
        tstop(:,i) = ts_min
-       dtdrag = 0.9*ts_min
+       dtdrag = C_force*ts_min
     endif

 #ifdef IND_TIMESTEPS
