diff --git a/src/main/readwrite_dumps_hdf5.F90 b/src/main/readwrite_dumps_hdf5.F90
index 9623c816..a1e28ade 100644
--- a/src/main/readwrite_dumps_hdf5.F90
+++ b/src/main/readwrite_dumps_hdf5.F90
@@ -467,7 +467,7 @@ end subroutine write_dump
 !+
 !-------------------------------------------------------------------
 subroutine read_dump(dumpfile,tfile,hfactfile,idisk1,iprint,id,nprocs,ierr,headeronly,dustydisc)
- use boundary,       only:xmin,xmax,ymin,ymax,zmin,zmax
+ use boundary,       only:set_boundary
  use dim,            only:maxp,gravity,maxalpha,mhd,use_dust,use_dustgrowth, &
                           h2chemistry,store_temperature,nsinkproperties,maxp_hard
  use eos,            only:ieos,polyk,gamma,polyk2,qfacdisc,isink
@@ -479,7 +479,7 @@ subroutine read_dump(dumpfile,tfile,hfactfile,idisk1,iprint,id,nprocs,ierr,heade
                           nptmass,xyzmh_ptmass,vxyz_ptmass,ndustlarge,        &
                           ndustsmall,grainsize,graindens,Bextx,Bexty,Bextz,   &
                           alphaind,poten,Bxyz,Bevol,dustfrac,deltav,dustprop, &
-                          tstop,St,temperature,abundance,ndusttypes
+                          tstop,St,temperature,abundance,ndusttypes,periodic
 #ifdef IND_TIMESTEPS
  use part,           only:dt_in
 #endif
@@ -497,6 +497,7 @@ subroutine read_dump(dumpfile,tfile,hfactfile,idisk1,iprint,id,nprocs,ierr,heade
  logical, optional, intent(in)  :: dustydisc
 
  real(kind=4), allocatable :: dtind(:)
+ real :: xmin,xmax,ymin,ymax,zmin,zmax
 
  character(len=200) :: fileident
  integer :: errors(5)
@@ -597,6 +598,10 @@ subroutine read_dump(dumpfile,tfile,hfactfile,idisk1,iprint,id,nprocs,ierr,heade
  endif
 #endif
 
+ if (periodic) then
+    call set_boundary(xmin,xmax,ymin,ymax,zmin,zmax)
+ endif
+
 #ifdef ISOTHERMAL
  isothermal = .true.
 #else
@@ -668,7 +673,7 @@ end subroutine read_dump
 !+
 !-------------------------------------------------------------------
 subroutine read_smalldump(dumpfile,tfile,hfactfile,idisk1,iprint,id,nprocs,ierr,headeronly,dustydisc)
- use boundary,       only:xmin,xmax,ymin,ymax,zmin,zmax
+ use boundary,       only:set_boundary
  use dim,            only:maxp,gravity,maxalpha,mhd,use_dust,use_dustgrowth, &
                           h2chemistry,store_temperature,nsinkproperties
  use eos,            only:ieos,polyk,gamma,polyk2,qfacdisc,isink
@@ -680,7 +685,8 @@ subroutine read_smalldump(dumpfile,tfile,hfactfile,idisk1,iprint,id,nprocs,ierr,
                           nptmass,xyzmh_ptmass,vxyz_ptmass,ndustlarge,      &
                           ndustsmall,grainsize,graindens,Bextx,Bexty,Bextz, &
                           alphaind,poten,Bxyz,Bevol,dustfrac,deltav,        &
-                          dustprop,tstop,St,temperature,abundance,ndusttypes
+                          dustprop,tstop,St,temperature,abundance,ndusttypes, &
+                          periodic
 #ifdef IND_TIMESTEPS
  use part,           only:dt_in
 #endif
@@ -701,6 +707,7 @@ subroutine read_smalldump(dumpfile,tfile,hfactfile,idisk1,iprint,id,nprocs,ierr,
  logical, optional, intent(in)  :: dustydisc
 
  real(kind=4) :: dtind(npart)
+ real :: xmin,xmax,ymin,ymax,zmin,zmax
 
  character(len=200) :: fileident
  integer :: errors(5)
@@ -790,6 +797,10 @@ subroutine read_smalldump(dumpfile,tfile,hfactfile,idisk1,iprint,id,nprocs,ierr,
  call allocate_memory(int(npart / nprocs))
 #endif
 
+ if (periodic) then
+    call set_boundary(xmin,xmax,ymin,ymax,zmin,zmax)
+ endif
+
 #ifdef ISOTHERMAL
  isothermal = .true.
 #else
