diff --git a/src/main/force.F90 b/src/main/force.F90
index 8e434950..26f572d8 100644
--- a/src/main/force.F90
+++ b/src/main/force.F90
@@ -3006,7 +3006,7 @@ subroutine finish_cell_and_store_results(icall,cell,fxyzu,xyzh,vxyzu,poten,dt,dv
        tstop(:,i) = tstopi(:)
     elseif (use_dust .and. .not.use_dustfrac) then
        tstop(:,i) = ts_min
-       dtdrag = 0.9*ts_min
+       dtdrag = 20*C_force*ts_min
     endif
 
     if (do_radiation.and.iamgasi) then
